spices/0002-uri-rewriting-for-mirroring.adoc

```diff
@@ -0,0 +1,319 @@
+= SPICE 2: URI Rewriting for Mirroring
+
+* Proposal: link:./0002-uri-rewriting-for-mirroring.adoc[SPICE-0002]
+* Author: link:https://github.com/holzensp[Philip Hölzenspies]
+* Status: Review
+* Implemented in: Pkl 0.26.0
+* Category: Tooling
+
+== Introduction
+
+This SPICE discusses rewrite rules for URIs of modules being `import`ed, so as to use mirrored alternatives.
```

Comment:
This is a scary feature. I’m worried that it might cause more harm than good. Does Pkl do “feature previews” like (say) Java? Does this feature exist in other languages, especially in languages with a similar safety mindset?

Reply:
This is naturally an opt-in feature. Pkl would not do any rewriting unless specifically configured to do so. Unless you were thinking of something else? E.g. a library author being able to opt out of rewrites?